### PR TITLE
 Port CA2213 (Disposable fields should be disposed) to System.Runtime.Analyzers

### PR DESCRIPTION
**Pull request by srivatsn**
This involves making the fixer go through SyntaxGenerator instead of having  C#\VB specific fixers. A couple of cases weren't covered by the fixer before like for example dispatching through the interface if the Dispose method cannot be referenced by name (Explicit impl). That scenario wasn't being understood by the analyzer either. I've fixed that. Note that the analyzer needs flow analysis to be fully correct and so only covers some straightforward scenarios.

### CODE REVIEW

**Comment by srivatsn** [comment]
@shyamnamboodiripad @mavasani @tmeschter @heejaechang @jmarolf - can you please take a look?

File: src/Diagnostics/FxCop/System.Runtime.Analyzers/VisualBasic/Usage/DisposableFieldsShouldBeDisposed.vb [hunk 1]
```diff
@@ -29,19 +28,35 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.FxCopAnalyzers.Usage
                     Case SyntaxKind.SimpleMemberAccessExpression
                         ' NOTE: This cannot be optimized based on memberAccess.Name because a given method
                         ' may be an explicit interface implementation of IDisposable.Dispose.
+                        ' If the right hand side of the member access binds to IDisposable.Dispose
```
Comment (mavasani) [inline src/Diagnostics/FxCop/System.Runtime.Analyzers/VisualBasic/Usage/DisposableFieldsShouldBeDisposed.vb]: Seems like lot of code in this method is very similar to the that in C# analyzer, except for just few language specific syntax types. We can probably refactor this to share more code, if you think that's worthwhile :)

**Comment by mavasani** [comment]
:+1:

**Comment by srivatsn** [comment]
@dotnet-bot  test this please

**Comment by srivatsn** [comment]
@dotnet-bot test this please

**Comment by srivatsn** [comment]
One more review please?

**Comment by heejaechang** [comment]
:+1:

**Pull request merged**